homeassistant/components/template/config.py

```diff
@@ -382,11 +386,11 @@ async def async_validate_config(hass: HomeAssistant, config: ConfigType) -> Conf
             definitions = (
                 list(template_config[new_key]) if new_key in template_config else []
             )
-            definitions.extend(
-                rewrite_legacy_to_modern_configs(
-                    hass, new_key, template_config[old_key], legacy_fields
-                )
-            )
+            for definition in rewrite_legacy_to_modern_configs(
+                hass, new_key, template_config[old_key], legacy_fields
+            ):
+                create_legacy_template_issue(hass, definition, new_key)
+                definitions.append(definition)
```

Comment:
I think we should add cleanup of issues which are no longer there when the configuration is (re)loaded.

My suggestion would be:
- Store the issue id of all created issues when parsing the configuration
- Iterate over the template issues in the issue registry, issues which have a matching translation_key or data but not a matching issue id can be removed

If we need to keep apart issues which are created by legacy templates under the platform domain from legacy templates under the `template` domain, we can keep track of that by adding something to the issue data.